pkg/utils/config/main.go

```diff
@@ -0,0 +1,129 @@
+	nriPluginKey = "io.containerd.nri.v1.nri"
+	disableKey   = "disable"
+)
+
+func main() {
+	// Load TOML content
+	tomlData, err := os.ReadFile(tomlFilePath)
+	if err != nil {
+		log.Fatalf("Error reading TOML file: %v", err)
+	}
+
+	tomlMap, err := unmarshalTOML(tomlData)
+	if err != nil {
+		log.Fatalf("Error unmarshalling TOML: %v", err)
+	}
+
+	tomlMap = updateNRIplugin(tomlMap)
+
+	buf, err := addIndentation(tomlMap)
+	if err != nil {
+		log.Fatalf("Error adding indentation: %v", err)
+	}
+
+	if err := writeToFile(tomlFilePath, buf.String()); err != nil {
+		log.Fatalf("Error writing to file: %v", err)
+	}
+
+	if err := restartContainerd(); err != nil {
+		log.Fatalf("Failed to restard the containerd: %v", err)
+	}
+}
+
+// restartContainerd uses nsenter to access the host's mount and PID namespaces.
+// This enables us to restart the host's containerd from within the init container
+// after making necessary configuration modifications.
+func restartContainerd() error {
+	command := "nsenter"
+	args := []string{"-t", "1", "-m", "systemctl", "restart", "containerd"}
+
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err := cmd.Run()
+	if err != nil {
+		fmt.Printf("Error executing nsenter command: %v\n", err)
+		return err
+	}
+	return nil
+}
+
+func unmarshalTOML(data []byte) (map[string]interface{}, error) {
+	// Unmarshal TOML content into a map
+	var tomlMap map[string]interface{}
+	if err := tomlv2.Unmarshal(data, &tomlMap); err != nil {
+		return nil, err
+	}
+	return tomlMap, nil
+}
+
+// go-tomml v2 doesn't automatically indent content of tables by default.
+// Encoder.SetIndentTables can be used for that.
+func addIndentation(config map[string]interface{}) (bytes.Buffer, error) {
+	buf := bytes.Buffer{}
+	enc := tomlv2.NewEncoder(&buf)
+	enc.SetIndentTables(true)
+	if err := enc.Encode(config); err != nil {
+		return bytes.Buffer{}, err
+	}
+	return buf, nil
+}
+
+func writeToFile(filePath, content string) error {
+	f, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	w := bufio.NewWriter(f)
+	_, err = w.WriteString(content)
+	if err != nil {
+		return err
+	}
+	return w.Flush()
+}
+
+func updateNRIplugin(config map[string]interface{}) map[string]interface{} {
+	if plugins, exists := config["plugins"].(map[string]interface{}); exists {
+		if nri, ok := plugins[nriPluginKey].(map[string]interface{}); ok {
+			if _, ok := nri["disable"].(bool); ok {
+				nri[disableKey] = false
+				log.Println("Enabled NRI...")
+			} else {
+				nri[disableKey] = false
+				log.Println("Added 'disable' field to enable NRI...")
+			}
+		} else {
+			plugins[nriPluginKey] = map[string]interface{}{
+				disableKey: false,
+			}
+			log.Println("NRI plugin section not found, adding it to enable NRI...")
+		}
+	} else {
+		fmt.Println("Top level plugins section not found, adding it to enable NRI...")
+		config = make(map[string]interface{})
```

Comment:
Isn't this resetting the configuration completely instead of just adding a "plugins" section with the NRI plugin enabled ?

Reply:
I don't think so. Let me check and confirm.

Reply:
I tried with some data unrelated to plugins as in the example and in that case yes. You are right. I will fix it.
```
[debug]
  address = "/run/containerd/debug.sock"
  uid = 0
  gid = 0
  level = "info"

[metrics]
  address = ""
  grpc_histogram = false

[cgroup]
  path = ""

```